### PR TITLE
Add VideoRangeType to video conditions

### PR DESCRIPTION
**Pull request by nyanmisaka**
This is used to distinguish whether the client supports specific VideoRangeType, such as SDR, HDR10, HLG and DOVI.
**Since it's basically a copy-paste from the Video Profile condition, usages are similar.**

@dmitrylyzo has tried to add the "IsHdr" condition in https://github.com/jellyfin/jellyfin/pull/7199, but I think that's isn't enough for various HDR types mentioned above.

With this PR, clients can declare supported video range types by querying device attributes. Server will follow the condition from clients to make a right decision between using Transcoding, DirectPlay, DirectStreaming or Remuxing.

**Web**
https://github.com/jellyfin/jellyfin-web/pull/3713

**Changes**
- Add VideoRangeType to video conditions

**Issues**
Should fixes https://github.com/jellyfin/jellyfin-roku/issues/552 if roku client implement this.

### CODE REVIEW

File: MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs [hunk 1]
```diff
@@ -1753,6 +1753,21 @@ public bool CanStreamCopyVideo(EncodingJobInfo state, MediaStream videoStream)
...
+            var requestedRangeTypes = state.GetRequestedRangeTypes(videoStream.Codec);
+            if (requestedProfiles.Length > 0)
+            {
+                if (string.IsNullOrEmpty(videoStream.VideoRangeType))
+                {
+                    return false;
+                }
+
+                if (!string.IsNullOrEmpty(videoStream.VideoRangeType)
+                    && !requestedRangeTypes.Contains(videoStream.VideoRangeType, StringComparison.OrdinalIgnoreCase))
```
Comment (crobibero) [inline MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs:1765]:
```suggestion
                if (!requestedRangeTypes.Contains(videoStream.VideoRangeType, StringComparison.OrdinalIgnoreCase))
```

`videoStream.VideoRangeType` will always have a value here

File: MediaBrowser.Model/Entities/MediaStream.cs [hunk 2]
```diff
@@ -104,32 +104,23 @@ public string VideoRange
         {
             get
             {
-                if (Type != MediaStreamType.Video)
-                {
-                    return null;
-                }
-
-                var colorTransfer = ColorTransfer;
+                var (videoRange, videoRangeType) = getVideoColorRange();
```
Comment (crobibero) [inline MediaBrowser.Model/Entities/MediaStream.cs:107]:
```suggestion
                var (videoRange, _) = getVideoColorRange();
```

File: MediaBrowser.Model/Entities/MediaStream.cs [hunk 3]
```diff
@@ -104,32 +104,23 @@ public string VideoRange
...
-                }
+        /// <summary>
+        /// Gets the video range type.
+        /// </summary>
+        /// <value>The video range type.</value>
+        public string VideoRangeType
+        {
+            get
+            {
+                var (videoRange, videoRangeType) = getVideoColorRange();
```
Comment (crobibero) [inline MediaBrowser.Model/Entities/MediaStream.cs:121]:
```suggestion
                var (_, videoRangeType) = getVideoColorRange();
```

File: MediaBrowser.Model/Entities/MediaStream.cs [hunk 4]
```diff
@@ -571,5 +562,39 @@ public bool SupportsSubtitleConversionTo(string toCodec)
 
             return true;
         }
+
+        public (string VideoRange, string VideoRangeType) getVideoColorRange()
```
Comment (crobibero) [inline MediaBrowser.Model/Entities/MediaStream.cs:566]:
```suggestion
        public (string VideoRange, string VideoRangeType) GetVideoColorRange()
```

Functions should begin with an uppercase letter

**Review by crobibero — APPROVED** [review]

File: MediaBrowser.Model/Dlna/StreamInfo.cs [hunk 5]
```diff
@@ -280,6 +280,29 @@ public string TargetVideoProfile
...
+        {
+            get
+            {
+                if (IsDirectStream)
+                {
+                    return TargetVideoStream?.VideoRangeType;
+                }
+
+                var targetVideoCodecs = TargetVideoCodec;
+                var videoCodec = targetVideoCodecs.FirstOrDefault();
```
Comment (cvium) [inline MediaBrowser.Model/Dlna/StreamInfo.cs:296]:
```suggestion
                var targetVideoCodecs = TargetVideoCodec;
                var videoCodec = targetVideoCodecs.Length == 0 ? null : targetVideoCodecs[0];
```

**Review by cvium — APPROVED** [review]

**Pull request merged**